### ISSUE TITLE
LFO Reporter: Facilities Table Page

### ISSUE THREAD

**Issue by pbastia** [bcgov/cas-reporting#312]
#### Description:

[Wireframe + flow here
](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=16744-19775&t=54jOo0gFSFd18imr-1)

![Image](https://github.com/user-attachments/assets/7e8db8f1-e8b6-4b93-a81f-92605b73fa69)

#### Acceptance Criteria:

**NOTE to Devs: The "actions" column on this table should act the same was as the one on the "Reporting Dashboard". 
See tickets #274 and #296 and #219 

Given I am an LFO industrial reporter who has reviewed my list of facilities (#197 )
When I click continue from the "review facilities" page
Then I am brought to the facility table page, where all the facilities I had previously selected are displayed with the information as shown in the wireframe

Given that I am an LFO industrial reporter on the Facility table page
When I click on "Start" in the actions column for a facility
Then I am brought to "Review Facility Information" page for that facility and a report has been started for that facility. 

#### Development Checklist:

- [x] Add a facility table page at /reports/{version}/report-information that displays facilities selected on the "review facilities" page.
- [x] Display facility data from the "Review Facilities" page with columns: Facility Name, Facility BCGHG ID, Status, and Actions.
- [x] Implement search for each column: Facility Name and  Facility BCGHG ID
- [x] The Value in "Actions" column will be based on completed status, if is completed then show "View Details" otherwise "Continue"
- [x] Store and update facility report status Completed(checkbox),  Add a field "is_completed" in facility_report table, boolean type
- [x] On clicking Continue/View Details, navigate to the Review Facility Information page
- [x] Continue button is disabled if all the facilitie are not marked as complted, if all the completed, continue button will redirect to additional information.
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Blocked By 
#197 

#### Blocking 
#320 

#### Notes:

-

**Comment by suhafa** [comment]
Wireframe added

**Comment by rdromey** [comment]
@ayeshmcg to take a first pass at this

**Comment by ayeshmcg** [comment]
@patriciarussellCAS @zoeyli-46 
cc: @dleard 
I'm unsure about the approach we'll use to capture facility status.
**Suggestion:**
We could consider adding a "Status" field with a checkbox option. This would allow users to manually mark facilities they have completed.
We can discuss this further to finalize the approach.

**Comment by patriciarussellCAS** [comment]
Thanks @ayeshmcg - I think we flagged this for a dev/design sync. 
From my last conversation with others, I think we landed on only having 2 statuses: Not Started and Draft. It is not complete until the whole report is submitted. But let's talk next week :)

**Comment by ayeshmcg** [comment]
@patriciarussellCAS , @zoeyli-46 
cc: @dleard 
I have updated the Dev checklist as per our discussion

**Comment by patriciarussellCAS** [comment]
Hey @shon-button or @ayeshmcg - can you let me know if you think this will be done in the next day or two? We'll need this to review #320 so want to provide an update to Tammie who is planning UAT. 
Thanks!

**Comment by ayeshmcg** [comment]
@patriciarussellCAS  I’m currently working on the changes suggested by @shon-button , and I expect to complete them within a day.

**Comment by ayeshmcg** [comment]
To test in dev:-
https://cas-bciers-frontend-dev.apps.silver.devops.gov.bc.ca/reporting/reports/2/facilities/report-information

**Comment by fviduya** [comment]
I have tested this in the DEV environment, [here is the test document for reference](https://bcgov-my.sharepoint.com/:w:/g/personal/francesca_viduya_gov_bc_ca/EYroyNaBxAFGvnar4qjFS7UBUG3uiOyAXf-8n-eDgEcCMA?e=kRTQh6).

I can verify that:
- [x] When I click on ```Continue``` from the Review Facilities page, the page redirects to the Report Information page where facilities I have selected from the Review Facilities page are displayed
- [x] When a report for a facility has not been completed yet, the action ```Continue``` is clickable under the Actions column
- [x] When I click on ```Continue``` button from the Actions column, the page redirects to the Review Facility Information page
- [x] When I click on ```Complete``` checkbox, the Actions column for that facility changes to ```View Details``` 
- [x] When I click on ```View Details``` button from the Actions column, the page redirects to the Review Facility Information page
- [x] The ```Save & Continue``` button in the Reports Information page is disabled when there is at least one facility that has not completed a report yet
- [x] When all facilities have completed reporting (```Complete``` is checked under Status column), the ```Save & Continue``` button is enabled  

Hi @patriciarussellCAS / @vesselak, ~I wanted to get your thoughts on this. Currently, the statuses are not updating based on the progress of the report, and the default status I see is ```Completed``` with a check box where you can manually check it, and the default for Actions column is ```Continue``` where in reporting we use that whenever the report has been started but not submitted yet. I am unsure whether to include the Statuses and state of Action on this ticket or create another issue for that and make this ticket as just for navigation (because that's working here).  Moving to Pending PO for review of test and notes. Thank you!
*Will not be adding the QA tested label yet and will wait for confirmation for these ^~ I've met with @ayeshmcg to talk about the Statuses on this and the expectations as well, the status here is just the ```Complete``` checkbox which the user will manually check once they complete the report for that facility.. Moving to Pending PO for review of test and notes. Thank you!

**Comment by patriciarussellCAS** [comment]
Reviewed. Moving to Accepted.
